### DISCUSSION TITLE
Can Oracledb._Error show the error location in "Lines and Columns"?

### DISCUSSION THREAD

**Discussion by snickerjp** [oracle/python-oracledb#214] [Q&A] [answered]
Can Oracledb._Error show the error location in "Lines and Columns"?

In case of SQL errors
Can I get an error message "Line:X and Column:Y" like SQL*Plus?

I have checked the documentation,
* the error offset.
* the Oracle error number (ORA-XXXXX).
* the Oracle message of the error.

can be retrieved, but it does not seem to be able to receive messages like "Line:X and Column:Y".

Also, I think it is possible to calculate from Offset, but is there already such a function available?

Reference
https://python-oracledb.readthedocs.io/en/latest/api_manual/module.html#oracledb-error-objects

* example error message screen shot (SQLcl)
![image](https://github.com/oracle/python-oracledb/assets/1247622/e8b9bca8-4517-4dff-ad10-46e18bd5858a)

**Comment by snickerjp** [comment]
To solve the problem, I wrote the following code in python.

However, if it already exists, I would like to use an existing one.

```python
   def calculate_line_and_column(query, offset):
        lines = query.split("\n")
        total_offset = 0

        for line_num, line in enumerate(lines, start=1):
            total_offset += len(line) + 1  # Add 1 for the newline character
            if total_offset > offset:
                column_num = offset - (total_offset - len(line))
                return line_num, column_num
```

**Reply by anthony-tuininga** [reply, marked as answer]
There is no method currently to calculate the line and column from the offset. If you would like to see this included in python-oracledb, please create an enhancement issue from this discussion.

**Reply by snickerjp** [reply]
Thank you for your response.

I will consider submitting an issue.

**Reply by SteveOgh** [reply]
Thank you [snickerjp](https://github.com/snickerjp) !  This is exactly what I needed, hard to believe it is not already there.
